src/main/java/seedu/address/model/transaction/BuyTransaction.java

```diff
@@ -25,4 +25,24 @@ public String toString() {
         return "You bought " + quantity + " quantity of "
                 + goods + " at " + price + " each";
     }
+
+    /**
+     * Returns true if both transactions have the same identity and data fields.
+     * This defines a stronger notion of equality between two transactions.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof BuyTransaction)) {
+            return false;
+        }
```

Comment:
I think this part is testable, but isn't covered by a test yet (as warned by codecov). Perhaps, it might be better if we could add such a case in?